packages/jest-reporters/src/GithubActionsReporter.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {AggregatedResult, TestResult} from '@jest/test-result';
+import BaseReporter from './BaseReporter';
+import type {Context} from './types';
+
+const newLine = /\n/g;
+const encodedNewLine = '%0A';
+const lineAndColumnInStackTrace = /^.*?:([0-9]+):([0-9]+).*$/;
+
+export default class GithubActionsReporter extends BaseReporter {
+  onRunComplete(
+    _contexts?: Set<Context>,
+    aggregatedResults?: AggregatedResult,
+  ): void {
+    if (!process.env.GITHUB_ACTIONS) {
+      return;
+    }
+
+    const messages = getMessages(aggregatedResults?.testResults);
+
+    for (const message of messages) {
+      this.log(message);
+    }
+  }
+}
+
+function getMessages(results: Array<TestResult> | undefined) {
+  if (!results) return [];
+
+  return results.reduce(
+    flatMap(({testFilePath, testResults}) =>
+      testResults
+        .filter(r => r.status === 'failed')
+        .reduce(
+          flatMap(r => r.failureMessages),
+          [],
+        )
+        .map(m => m.replace(newLine, encodedNewLine))
```

Comment:
There's 3 characters that need to be escaped here. This is what the [action's toolkit](https://github.com/actions/toolkit/blob/fcb8c4ca798bd0d81a9b07e2f7dffee8397c4707/packages/core/src/command.ts#L80-L85) is doing.

```typescript
function escapeData(s: any): string {
  return toCommandValue(s) // ensures the value is a string
    .replace(/%/g, '%25')
    .replace(/\r/g, '%0D')
    .replace(/\n/g, '%0A')
}
```

Reply:
Should we depend on it?

Reply:
There's a helper to create the annotation command in the toolkit, but it might be overkill to depend on here with how many downloads this gets per week. The main thing is to make sure each value going into the command is escaped which is just those 3 characters.